src/operations/UpdateContext.ts

```diff
@@ -1,11 +1,17 @@
-import { Operation, createContext } from "effection";
+import { Operation, createContext, lift } from "effection";
 import { LoaderState } from "../hooks/useLoader";
 
-type Update = (state: LoaderState<any>) => Operation<void>;
+type Update<T> = (state: LoaderState<T>) => Operation<void>;
 
-export const UpdateContext = createContext<Update>("update");
+export const UpdateContext = createContext<Update<unknown>>("update");
 
-export function* update<T>(state: LoaderState<T>): Operation<void> {
-  let setState = yield* UpdateContext;
-  yield* setState(state);
+export const update: Update<unknown> = function* update(value) {
+  const setState = yield* UpdateContext;
+  yield* setState(value);
 }
+
+export function* setUpdate(setState: (value: LoaderState<unknown>) => void): Operation<Update<unknown>> {
+  const lifted = lift(setState);
+  yield* UpdateContext.set(lifted);
+  return lifted;
```

Comment:
I wouldn't return `lifted` and instead make `setUpdate()` return `Operation<void>`. There is already an official way for using `update()`, it's the named `update()` operation.

Reply:
Oh yeah, I forgot that it's coming from context.